compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -11358,49 +11359,49 @@ private STNode parseTupleMemberTypeDescList() {
         }
 
         // Parse first typedesc, that has no leading comma
-        STNode typeDesc = parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_TUPLE);
-
+        STNode typeDesc = parseMemberDescriptor();
         return parseTupleTypeMembers(typeDesc, typeDescList);
     }
 
-    private STNode parseTupleTypeMembers(STNode typeDesc, List<STNode> typeDescList) {
-        typeDesc = parseComplexTypeDescriptor(typeDesc, ParserRuleContext.TYPE_DESC_IN_TUPLE, false);
-
-        STNode tupleMemberRhs = parseTypeDescInTupleRhs();
-        if (tupleMemberRhs != null) {
-            typeDesc = STNodeFactory.createRestDescriptorNode(typeDesc, tupleMemberRhs);
-        }
-
+    private STNode parseTupleTypeMembers(STNode membertypeDesc, List<STNode> memberTypeDescList) {
```

Comment:
to about confusion with "member type desc" definition
```suggestion
    private STNode parseTupleTypeMembers(STNode firstMember, List<STNode> memberList) {
```